### PR TITLE
Fix @appwrite#3600 - Binding parameters issues with hyphen in `mariaDB`

### PR DESCRIPTION
**Pull request by iifawzi**
## What does this PR do?
Hi, this PR is addressing the issues of using `hyphen` while binding parameters in `mariaDB` 
Fixes https://github.com/appwrite/appwrite/issues/3600

As per [PDO Source Code](https://github.com/php/php-src/blob/5638fafa665899523861a12966b42a9c609a1664/ext/pdo/pdo_sql_parser.re#L48), parameters variables names can only be alphanumeric characters plus underscore.

I've also used `back ticks` for the db columns to be parsed correctly by the db, since hyphen usually if not always, causing issues when back ticks are not used. 

## Test Plan
Tests are added to `MySQLTest` to test the `sum`, `count` and `findOne` functionalities.
Signed-off-by: iifawzi <iifawzie@gmail.com>

### CODE REVIEW

File: src/Database/Adapter/MariaDB.php [hunk 1]
```diff
@@ -993,7 +994,8 @@ public function count(string $collection, array $queries = [], int $max = 0): in
         foreach ($queries as $i => $query) {
             if ($query->getOperator() === Query::TYPE_SEARCH) continue;
             foreach ($query->getValues() as $key => $value) {
-                $stmt->bindValue(':attribute_' . $i . '_' . $key . '_' . $query->getAttribute(), $value, $this->getPDOType($value));
+                $attribute = preg_replace("/[^\w]+/", "_", $query->getAttribute());
```
Comment (iifawzi) [inline src/Database/Adapter/MariaDB.php:997]: The same, but the colon will also be replaced, if present.

**Review by iifawzi — COMMENTED** [review]
The regex parts

File: src/Database/Adapter/MariaDB.php [hunk 2]
```diff
@@ -890,7 +890,8 @@ public function find(string $collection, array $queries = [], int $limit = 25, i
         foreach ($queries as $i => $query) {
             if ($query->getOperator() === Query::TYPE_SEARCH) continue;
             foreach ($query->getValues() as $key => $value) {
-                $stmt->bindValue(':attribute_' . $i . '_' . $key . '_' . $query->getAttribute(), $value, $this->getPDOType($value));
+                $attribute = preg_replace("/[^:\w]+/", "_", $query->getAttribute());
```
Comment (iifawzi) [inline src/Database/Adapter/MariaDB.php:893]: `/[^:\w]+/` replaces all non colon and non word characters with `_`

**Comment by iifawzi** [comment]
Hi, is there any updates regarding this?

**Comment by stnguyen90** [comment]
@iifawzi, looks like there is a merge conflict now. Would you please try to resolve it?

**Comment by iifawzi** [comment]
> @iifawzi, looks like there is a merge conflict now. Would you please try to resolve it?

All conflicts have been resolved, I've also updated the code according to the `Query` class changes.

**Comment by iifawzi** [comment]
isn't it easier to merge a pull request, than `copying` the changes and committing them in separate commits?

**Pull request closed without merging**